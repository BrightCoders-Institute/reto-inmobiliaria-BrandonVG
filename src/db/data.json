[
  {
    "id": 1,
    "name": "The Willows",
    "address": "3517 W. Gray St. Utica",
    "rooms": "3",
    "bathrooms": "2",
    "size": "230",
    "price": "440",
    "stars": "4.7",
    "like": false,
    "img": "https://definicion.de/wp-content/uploads/2011/01/casa-2.jpg"
  },
  {
    "id": 2,
    "name": "Cozy Cottage",
    "address": "1234 Main St. Anytown",
    "rooms": "2",
    "bathrooms": "1",
    "size": "120",
    "price": "200",
    "stars": "4.2",
    "like": true,
    "img": "https://www.bbva.com/wp-content/uploads/2021/04/casas-ecolo%CC%81gicas_apertura-hogar-sostenibilidad-certificado-.jpg"
  },
  {
    "id": 3,
    "name": "Luxury Villa",
    "address": "9876 Elm St. Rivertown",
    "rooms": "5",
    "bathrooms": "3",
    "size": "400",
    "price": "800",
    "stars": "4.9",
    "like": true,
    "img": "https://images.adsttc.com/media/images/5d34/e507/284d/d109/5600/0240/large_jpg/_FI.jpg"
  },
  {
    "id": 4,
    "name": "Seaside Retreat",
    "address": "5678 Beach Blvd. Oceanville",
    "rooms": "4",
    "bathrooms": "2.5",
    "size": "300",
    "price": "600",
    "stars": "4.5",
    "like": false,
    "img": "https://img10.naventcdn.com/avisos/18/00/66/32/80/15/720x532/359290663.jpg"
  },
  {
    "id": 5,
    "name": "Mountain Cabin",
    "address": "2468 Pine St. Summitville",
    "rooms": "2",
    "bathrooms": "1",
    "size": "150",
    "price": "300",
    "stars": "4.3",
    "like": false,
    "img": "https://images.adsttc.com/media/images/623c/4fa0/3e4b/3145/3000/001b/large_jpg/_d_ambrosio_07._copy.jpg"
  },
  {
    "id": 6,
    "name": "Urban Loft",
    "address": "6789 Oak St. Cityville",
    "rooms": "1",
    "bathrooms": "1",
    "size": "80",
    "price": "150",
    "stars": "3.9",
    "like": true,
    "img": "https://i.blogs.es/c68014/casa-3d/840_560.jpeg"
  },
  {
    "id": 7,
    "name": "Rustic Farmhouse",
    "address": "1357 Maple Rd. Countryside",
    "rooms": "4",
    "bathrooms": "3",
    "size": "320",
    "price": "550",
    "stars": "4.6",
    "like": true,
    "img": "https://propiedadescancun.mx/wp-content/uploads/2019/09/fachadaR_casa_residencial_lagos_del_sol_cancun_ceibey5-7.jpg"
  },
  {
    "id": 8,
    "name": "Modern Apartment",
    "address": "7890 Elm St. Metropolitan",
    "rooms": "2",
    "bathrooms": "2",
    "size": "180",
    "price": "400",
    "stars": "4.4",
    "like": false,
    "img": "https://www.rocketmortgage.com/resources-cmsassets/RocketMortgage.com/Article_Images/Large_Images/TypesOfHomes/types-of-homes-hero.jpg"
  },
  {
    "id": 9,
    "name": "Countryside Retreat",
    "address": "2468 Willow Ln. Meadows",
    "rooms": "3",
    "bathrooms": "2",
    "size": "250",
    "price": "500",
    "stars": "4.8",
    "like": true,
    "img": "https://www.rocketmortgage.com/resources-cmsassets/RocketMortgage.com/Article_Images/Large_Images/TypesOfHomes/types-of-homes-hero.jpg"
  },
  {
    "id": 10,
    "name": "Charming Bungalow",
    "address": "1012 Oak St. Riverside",
    "rooms": "2",
    "bathrooms": "1.5",
    "size": "180",
    "price": "350",
    "stars": "4.2",
    "like": false,
    "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d6/George_L._Burlingame_House%2C_1238_Harvard_St%2C_Houston_%28HDR%29.jpg/1200px-George_L._Burlingame_House%2C_1238_Harvard_St%2C_Houston_%28HDR%29.jpg"
  }
]
