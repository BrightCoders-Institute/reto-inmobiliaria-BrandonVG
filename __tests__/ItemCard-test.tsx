import React from 'react';
import {create, act} from 'react-test-renderer';
import ItemCard from '../src/components/ItemCard';
import JsonReader from '../src/utils/JsonReader';
import data from '../src/db/data.json';
const inmuebles = JsonReader.jsonInmueblesToArray(data);

const tree = create(<ItemCard inmueble={inmuebles[0]} />);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

test('Like button pressed', () => {
  const likeButton = tree.root.findByProps({testID: 'likeButton'}).props;
  act(() => likeButton.onPress());
  const iconLike = tree.root.findByProps({testID: 'iconLike'}).props;
  expect(iconLike.color).toBe('red');
});
