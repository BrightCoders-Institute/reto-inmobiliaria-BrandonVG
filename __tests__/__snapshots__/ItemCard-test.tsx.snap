// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#F5FDFF",
      "borderRadius": 16,
      "height": 140,
      "marginBottom": 20,
      "width": "90%",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "margin": 15,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "alignItems": "center",
          "borderRadius": 10,
          "flex": 1,
          "justifyContent": "flex-end",
          "overflow": "hidden",
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://definicion.de/wp-content/uploads/2011/01/casa-2.jpg",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FBEDB7",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 25,
            "justifyContent": "center",
            "margin": 5,
            "width": 50,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#EEBA00",
                "fontSize": 11,
              },
              undefined,
              {
                "fontFamily": "Fontisto",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#856D35",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginLeft": 1,
            }
          }
        >
          4.7
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 2,
          "justifyContent": "space-between",
          "marginLeft": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#151525",
            "fontSize": 20,
            "fontWeight": "500",
          }
        }
      >
        The Willows
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontSize": 20,
              },
              {
                "marginRight": 10,
              },
              {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 14,
            }
          }
        >
          3517 W. Gray St. Utica
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "gray",
                "fontSize": 20,
              },
              {
                "marginRight": 10,
              },
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "color": "#151525",
                "fontWeight": "bold",
              },
              {
                "fontSize": 16,
                "marginRight": 15,
              },
            ]
          }
        >
          3
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "gray",
                "fontSize": 20,
              },
              {
                "marginRight": 10,
              },
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󱠙
        </Text>
        <Text
          style={
            [
              {
                "color": "#151525",
                "fontWeight": "bold",
              },
              {
                "fontSize": 16,
                "marginRight": 15,
              },
            ]
          }
        >
          2
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "gray",
                "fontSize": 20,
              },
              {
                "marginRight": 10,
              },
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰠡
        </Text>
        <Text
          style={
            [
              {
                "color": "#151525",
                "fontWeight": "bold",
              },
              {
                "fontSize": 16,
                "marginRight": 15,
              },
            ]
          }
        >
          230
           ft²
        </Text>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#151525",
                "fontWeight": "bold",
              },
              {
                "fontSize": 20,
              },
            ]
          }
        >
          $
          440
          /m
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": "#00b074",
              "borderRadius": 20,
              "height": 25,
              "justifyContent": "center",
              "marginRight": 15,
              "opacity": 1,
              "width": 25,
            }
          }
          testID="likeButton"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "white",
                  "fontSize": 15,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconLike"
          >
            󰣐
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
